cmd/sops/edit.go

```diff
@@ -135,18 +135,20 @@ func editTree(opts editOpts, tree *sops.Tree, dataKey []byte) ([]byte, error) {
 		return nil, common.NewExitError(fmt.Sprintf("Could not write output file: %s", err), codes.CouldNotWriteOutputFile)
 	}
 
-	// Close temporary file, since Windows won't delete the file unless it's closed beforehand
-	defer tmpfile.Close()
+	// Close the temporary file immediately
+	// On Windows, the standard library opens files in a way that prevents certain editors from writing to them.
+	tmpfile.Close()
+	var tmpfileName = tmpfile.Name()
```

Comment:
I can change everything back to using the existing `tmpfile` variable if required. I extracted the name because I felt it was better to stop using a closed file entirely (to prevent possible future confusion about the state of the file).